Implement Keyboard property in WinUI Entry

jsuarezruiz opened the pull request:
### Description of Change ###

Implement `Keyboard` property in WinUI Entry.

### PR Checklist ###

- [x] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)
- [x] Targets a single property for a single control (or intertwined few properties)
- [ ] Adds the property to the appropriate interface
- [x] Avoids any changes not essential to the handler property
- [x] Adds the mapping to the PropertyMapper in the handler
- [x] Adds the mapping method to the Android, iOS, and Standard aspects of the handler
- [x] Implements the actual property updates (usually in extension methods in the Platform section of Core)
- [x] Tags ported renderer methods with [PortHandler]
- [x] Adds an example of the property to the sample project (MainPage)
- [ ] Adds the property to the stub class
- [ ] Implements basic property tests in DeviceTests

#### Does this PR touch anything that might affect accessibility?
- No

jsuarezruiz committed:
Implement Keyboard property in WinUI Entry

rachelkang reviewed (dismissed):
@jsuarezruiz The keyboard type doesn't change :/

and build is failing :(

Redth committed:
Fix password NRE and use new WinUI Keyboard state api

Redth committed:
Remove now unnecessary current app use

rachelkang commented:
> @jsuarezruiz The keyboard type doesn't change :/
> 
> and build is failing :(

hmm I tested again, but it's the same :/

Redth committed:
Remove dup method

rachelkang commented:
still not working :(

rmarinho committed:
Merge branch 'main' into winui-entry-keyboard

rmarinho committed:
Merge branch 'main' into winui-entry-keyboard

Redth committed:
Consolidate all InputScope update logic

There was still an extension method which was updating the InputScope property independently of any others, so we had a last setter win situation.

This change ensures the UpdateInputScope is called whenever anything needs to ultimately set that property and accounts for multiple things which influence the set of input scope names that calculate the final value.

rachelkang reviewed (approved):
Builds and works great! Seeing the numeric keyboard on winui entry
